src/extension/wps-download/src/main/java/org/geoserver/wps/gs/download/DownloadAnimationProcess.java

```diff
@@ -194,16 +218,24 @@ public RawData execute(
                             return null;
                         });
         try {
-            for (Object parsedTime : parsedTimes) {
-                // turn parsed time into a specification and generate a "WMS" like request based on
-                // it
-                String mapTime = toWmsTimeSpecification(parsedTime);
+            for (int i = 0; i < totalTimes; i++) {
+                String mapTime = null;
+                String animateValue = null;
+                if (parsedTimes != null) {
+                    // turn parsed time into a specification and generate a "WMS" like request based
+                    // on it
+                    mapTime = toWmsTimeSpecification(parsedTimes.toArray()[i]);
+                } else {
+                    animateValue = animateValues.split(",")[i];
```

Comment:
Splitting over and over and over? Please prepare a list of parsed values while parsing (code above), something like:

```
List<Object> frameValues = new ArrayList<>();
if (time) {
  parse the times one by one
} else {
  split the animation params by comma
}
```